lib/web/apiserver.go

```diff
@@ -605,20 +663,122 @@ func (h *Handler) oidcLoginWeb(w http.ResponseWriter, r *http.Request, p httprou
 	return nil, nil
 }
 
-func (h *Handler) oidcLoginConsole(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
-	log.Debugf("oidcLoginConsole start")
-	var req *client.SSOLoginConsoleReq
+func (h *Handler) githubLoginWeb(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
+	logger := log.WithFields(log.Fields{trace.Component: "github"})
+	logger.Debug("Web login start")
+	clientRedirectURL := r.URL.Query().Get("redirect_url")
+	if clientRedirectURL == "" {
```

Comment:
was there a problem with random redirect_url that we had to check by stripping off the domain part and using relative vs absolute redirects?

Reply:
Yes, it is handled in the callback: 

https://github.com/gravitational/teleport/blob/roman/github/lib/web/apiserver.go#L750